OneCommunityGlobal/HGNRest
# Ilya - Consumables: Single Update

<img width="755" alt="Screenshot 2024-02-29 at 6 17 42 PM" src="https://github.com/OneCommunityGlobal/HGNRest/assets/105448137/ce62ea76-ad37-486a-8eeb-6df7fea2518b">

# Description
Adds functionality for the user to navigate to [/bmdashboard/consumables](http://localhost:3000/bmdashboard/consumables), select a consumable that needs to have it’s stock updated, press the pencil icon, enter how much of the consumable was used or wasted (in % or units) and press “Update Consumable”. The change should update the Consumables table right away. 

## Related PRS:
To test this backend PR you need to checkout the #2010 frontend PR.

## Main changes explained:
- New route created in bmConsumableRouter to handle post requests
- A function was added to bmConsumableController that handles “update” requests  
- Stock is updated based on units or percent
- Errors are sent properly to the front end 

## How to test:
1. check into current branch on the backend. Make sure you are using Node version 14.17.6
2. do **npm install** and **npm run build && npm start** to run this PR locally
3. checkout **ilya_consumables_single_update** branch on the front end, run commands **npm install** and **npm run start:local** to run on the front end
4. log in as any user
5. go to [http://localhost:3000/bmdashboard](http://localhost:3000/bmdashboard) and log in again
6. go to [http://localhost:3000/bmdashboard/consumables](http://localhost:3000/bmdashboard/consumables), select any item on the table, press the “pencil” icon, enter some numbers next to “Quantity Used” and/or “Quantity Wasted”. Select "%" as units or leave the item's default units 
7. Press “Update Consumable”
8. A success toast message should appear if everything is acceptable or a warning message if not
9. From the Consumables table, press “View” in the “Updates” column and verify that the change you made is reflected. The latest update should be on the bottom

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HGNRest/assets/105448137/c2d82e12-71e5-4618-867a-387754e90348



## Note:
This change will update the “buildingInventoryItems” collection on the database. As an alternative to step 9 you can open Compass, search the collection by the criteria {"__t":"consumable_item”} and verify that updateRecord property has a new entry with today’s date.
